app/modules/_common_tls.lua

```diff
@@ -254,7 +255,7 @@ function TLSSocket:_read(n)
         local i, o = self.out:pending()
         if i > 0 then
           net.Socket._write(self, self.out:read(), nil, function()
-            handshake()
+            timer.setImmediate(handshake)
```

Comment:
Is the `_write` callback sync sometimes?  Or is there some other reason to need to defer handshake?